internal/cmd/io_binary.go

```diff
@@ -0,0 +1,240 @@
+package cmd
```

Comment:
I'd like to see a brief description of what this file implements/exposes. The comment for `NewBinaryIO` could be fleshed out more also. What is a binary logger? What would it be used for? etc.